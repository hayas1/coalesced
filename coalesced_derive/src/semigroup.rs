use proc_macro2::TokenStream;
use syn::DeriveInput;

use crate::{constant::ConstantExt, semigroup::attr::ContainerAttr};

mod attr;
mod implementor;

pub fn gen_semigroup<C: ConstantExt>(derive: &DeriveInput) -> syn::Result<TokenStream> {
    let constant = C::constant();
    let attr = ContainerAttr::new(derive)?;

    todo!()
}
