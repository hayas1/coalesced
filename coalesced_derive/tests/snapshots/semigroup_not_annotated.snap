---
source: coalesced_derive/src/semigroup.rs
assertion_line: 77
expression: formatted
---
#[automatically_derived]
impl<T: std::ops::Add> Semigroup for UnnamedStruct<T> {
    fn semigroup_op(base: Self, other: Self) -> Self {
        Self {
            0: <coalesced::op::semigroup::add::Added<
                _,
            > as Construction<_>>::lift_op(base.0, other.0),
            1: <coalesced::op::annotation::replace::Replaced<
                _,
            > as Construction<_>>::lift_op(base.1, other.1),
        }
    }
}
